source/agora/consensus/EnrollmentPool.d

```diff
@@ -0,0 +1,374 @@
+/*******************************************************************************
+
+    Contains supporting code for managing enrollments registered by nodes
+    on the network to be a validator, which means the enrollment information
+    needs to be confirmed on the consensus protocol to be a validator. The
+    information is stored in a table of SQLite.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.EnrollmentPool;
+
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.consensus.data.Block;
+import agora.consensus.data.Enrollment;
+import agora.consensus.data.PreImageInfo;
+import agora.consensus.data.UTXOSet;
+import agora.consensus.Validation;
+import agora.utils.Log;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+mixin AddLogger!();
+
+/// Ditto
+public class EnrollmentPool
```

Comment:
There is an inconsistency in the API.

`hasEnrollment` doesn't throw, but `count` throws, `add` doesn't throw, but `remove` throws, and then `getEnrollment` throws and **also** returns a status flag.

These methods need to be consistent.

Reply:
> These methods need to be consistent.

Right, those functions should be consistent. I'll make it so.